### PR TITLE
Docs: Access control documentation reworked 2

### PR DESCRIPTION
**Pull request by comradekingu**
I got to where it says 4.
Keeping it short and sweet.
Beginning to trust what I am reading and making sense of the bigger picture now.

### CODE REVIEW

File: docs/admin/access.rst [hunk 1]
```diff
@@ -9,62 +9,52 @@ Access control
...
+In the permission system, roles define a set of permissions, and you can
+put users in groups that have one or more of these roles, described in
+:ref:`auth-model`.
 
 Site wide permissions
 +++++++++++++++++++++
 
-To manage permissions for a whole instance, just add users to `Users` (this is done
-by default using the :ref:`autogroup`), `Reviewers` and `Managers` groups. Keep
-all projects configured as `Public` (see :ref:`acl`).
```
Comment (comradekingu) [inline docs/admin/access.rst:39]: It isn't obvious _why_ all projects should be kept public?
Is it because permissions won't be site-wide if some projects are made private?

File: docs/admin/access.rst [hunk 2]
```diff
@@ -9,62 +9,52 @@ Access control
...
 
 .. include:: /snippets/not-hosted-libre.rst
 
 Members are granted any permissions assigned to groups they are in, so you can
-grant the user multiple permissions at once. Create groups and attach them to a project,
+grant a user multiple permissions at once. Create groups and attach them to a project,
 component, or language. You can put users in multiple groups, and permissions
 can overlap between them.
 
-Granting any selected permissions based on project, component or language
```
Comment (comradekingu) [inline docs/admin/access.rst:60]: This whole paragraph was the initial problem. I messed it up, and it wasn't good before that. I can't understand how it adds anything of value, so removed it…

File: docs/admin/access.rst [hunk 3]
```diff
@@ -9,62 +9,52 @@ Access control
...
-
-After installation a default set of groups are created, and you can use those
-to assign users roles for the whole instance (see :ref:`default-groups`). Additionally when
-:ref:`acl` is turned on, you can assign users to specific translation projects.
-More fine-grained configuration can be achieved using :ref:`custom-acl`.
-
-Common setups
--------------
-
-Locking down Weblate
```
Comment (comradekingu) [inline docs/admin/access.rst:27]: Moved this all the way to the bottom.
I don't see how locking down everything is common setup. There is rarely malice, and most instances don't run this. "Common setup → total lockdown" is a strange message to send IMO.

File: docs/admin/access.rst [hunk 4]
```diff
@@ -9,62 +9,52 @@ Access control
...
+put users in groups that have one or more of these roles, described in
+:ref:`auth-model`.
 
 Site wide permissions
 +++++++++++++++++++++
 
-To manage permissions for a whole instance, just add users to `Users` (this is done
-by default using the :ref:`autogroup`), `Reviewers` and `Managers` groups. Keep
-all projects configured as `Public` (see :ref:`acl`).
+To manage permissions for a whole instance, just add users to
```
Comment (comradekingu) [inline docs/admin/access.rst]:
```suggestion
```

Comment (comradekingu) [inline docs/admin/access.rst]:
```suggestion
```

File: docs/admin/access.rst [hunk 5]
```diff
@@ -9,62 +9,52 @@ Access control
...
+By default, new users get access to comment, provide suggestions, and make translations
```
Comment (comradekingu) [inline docs/admin/access.rst:25]: Guessing here.

File: docs/admin/access.rst [hunk 6]
```diff
@@ -9,62 +9,52 @@ Access control
...
+grant a user multiple permissions at once. Create groups and attach them to a project,
 component, or language. You can put users in multiple groups, and permissions
 can overlap between them.
 
-Granting any selected permissions based on project, component or language
-set. To achieve this, create a new group (e.g. `Czech translators`) and
-configure it for a given resource. Any assigned permissions will be granted to
-members of that group for selected resources.
+You can for example created `Czech translators` for
+as a means of granting write access to the Glossary for Czech and any projects
```
Comment (comradekingu) [inline docs/admin/access.rst:50]: This needs a better explanation of what things do and why.

File: docs/admin/access.rst [hunk 7]
```diff
@@ -79,46 +69,47 @@ Project access control
 .. note::
 
     By turning on access control, all users are prohibited from accessing anything
-    within a given project, unless you add the permissions for them to do just that.
+    within the given project, unless you add the permissions for them to do just that.
 
 .. include:: /snippets/not-hosted-libre.rst
 
 Limit user's access to individual projects by selecting a different access
-control variation on the :guilabel:`Access` tab in the :guilabel:`Settings` of
```
Comment (comradekingu) [inline docs/admin/access.rst:87]: Why variation? How many are there?

File: docs/admin/access.rst [hunk 8]
```diff
@@ -79,46 +69,47 @@ Project access control
...
+from the project page in question, as described in :ref:`manage-acl`.
 
 .. note::
 
     Even with access control turned on, some info will be available about your
     project:
 
-    * Statistics for the whole instance, including counts for all projects.
-    * Language summary for the whole instance, including counts for all projects.
+    * Statistics for the whole site, including the number of projects on it.
```
Comment (comradekingu) [inline docs/admin/access.rst:102]: What are counts? Tallies? People that live in castles?

File: docs/admin/access.rst [hunk 9]
```diff
@@ -79,46 +69,47 @@ Project access control
...
+    * Language summary for the whole site, including counts for all projects.
```
Comment (comradekingu) [inline docs/admin/access.rst:103]: Another "counts". Of?

File: docs/admin/access.rst [hunk 10]
```diff
@@ -79,46 +69,47 @@ Project access control
...
+    * The VCS project repository address, even if it is a private prepository.
```
Comment (comradekingu) [inline docs/admin/access.rst:104]: @nijel There was a person mentioning this in a bug-report I remember.

File: docs/admin/access.rst [hunk 11]
```diff
@@ -139,8 +130,7 @@ The authentication models consist of several objects:
     Individual permissions defined by Weblate. Permissions can not be
     assigned to users. This can only be done through assignment of roles.
 `Role`
-    A Role defines a set of permissions. This allows reuse of these sets in
-    several places, making the administration easier.
+    A role defines a set of permissions, and can be reused in several places.
```
Comment (comradekingu) [inline docs/admin/access.rst:132]:
```suggestion
    A role defines a set of permissions, for ease of use in several places.
```

File: docs/admin/access.rst [hunk 12]
```diff
@@ -165,9 +155,9 @@ The authentication models consist of several objects:
 Permission checking
 +++++++++++++++++++
 
-Whenever a permission is checked to decide whether one is able to perform a
-given action, the check is carried out according to scope, and the following
-checks are performed in this order:
+Whenever a permission is checked to decide whether the user is able to
+perform a given action, the check is carried out according to scope, and
```
Comment (comradekingu) [inline docs/admin/access.rst:158]: does the word "scope" have to be used?

File: docs/admin/access.rst [hunk 13]
```diff
@@ -214,7 +204,7 @@ Managing users and groups
 -------------------------
 
 All users and the various groups they are in can be managed using the
-Django admin interface available, which you can get to by appending
+Django admin-interface available, which you can get to by appending
```
Comment (comradekingu) [inline docs/admin/access.rst:206]:
```suggestion
available Django admin-interface, which you can get to by appending
```
as opposed to "many exist and one happened to be vacant". Bit redundant still.

File: docs/admin/access.rst (hunk 1 above)
Comment (Fat-Zer) [reply docs/admin/access.rst:39]: It's relevant for a "Common setup" "Site wide permissions" i.e. if you want to manage permissions site-wide you should keep projects as `public`.

File: docs/admin/access.rst [hunk 14]
```diff
@@ -9,62 +9,51 @@ Access control
...
 Site wide permissions
 +++++++++++++++++++++
 
-To manage permissions for a whole instance, just add users to `Users` (this is done
-by default using the :ref:`autogroup`), `Reviewers` and `Managers` groups. Keep
-all projects configured as `Public` (see :ref:`acl`).
+By default, new users get access to comment, provide suggestions, and make translations
+everywhere by being put in the `Users` group automatically by :ref:`autogroup`.
+You can put the users you deem fit in the `Reviewers` to let them review everything,
+and the `Managers` groups to do all of the above everywhere.
```
Comment (Fat-Zer) [inline docs/admin/access.rst:28]: The whole section looks irrelevant for the title: doesn't tell anything on how to manage the site-wide permissions.

Also it is misleading:
* `Reviewers` don't review everything, but just components configured to be reviewed.
* `Managers` has much more rights than listed above.

**Review by Fat-Zer — CHANGES_REQUESTED** [review]

**Comment by Fat-Zer** [comment]
The access control documentation needs **much** more love... But it's a step in a right direction...

File: docs/admin/access.rst [hunk 15]
```diff
@@ -510,3 +499,10 @@ however re-create them if you delete or rename them.
     Never remove the predefined Weblate groups and users, as this can lead to
     unexpected problems. If you have no use for them, you can removing all their
     privileges instead.
+
+Locking down Weblate
+++++++++++++++++++++
+
+If you experience problems with malice on your Weblate installation,
```
Comment (comradekingu) [inline docs/admin/access.rst:506]:
```suggestion
If you want ultimate control of who signs up for the platform, or experience problems with malice on your Weblate installation,
```
Unless this is true?

File: docs/admin/access.rst (hunk 10 above)
Comment (comradekingu) [reply docs/admin/access.rst]: Fixed spelling, needs answering.

File: docs/admin/access.rst [hunk 16]
```diff
@@ -79,46 +68,47 @@ Project access control
...
+each respective project. This automatically creates several groups for the
 project in question, see :ref:`groups`.
 
 :guilabel:`Access control` can be set to:
 
 Public
     Publicly visible, translatable for all logged-in users
 Protected
-    Publicly visible, but translatable only for selected users
+    Publicly visible, but only translatable for selected users
```
[the remark this answers is not in the document]
Comment (comradekingu) [reply docs/admin/access.rst:85]: Would you be happy with "only translatable for select users".

File: docs/admin/access.rst (hunk 7 above)
Comment (comradekingu) [reply docs/admin/access.rst:87]: Partly, but "variation" suggests it is a custom set of stuff that results in a variation, and "type" sounds like it is a list of pre-determined ways of control to chose from.

Adjust vs. pick.

File: docs/admin/access.rst [hunk 17]
```diff
@@ -79,46 +68,47 @@ Project access control
...
     By turning on access control, all users are prohibited from accessing anything
-    within a given project, unless you add the permissions for them to do just that.
+    within the given project, unless you add the permissions for them to do just that.
 
 .. include:: /snippets/not-hosted-libre.rst
 
 Limit user's access to individual projects by selecting a different access
-control variation on the :guilabel:`Access` tab in the :guilabel:`Settings` of
-each respective project.  This automatically creates several groups for the
+control type from the :guilabel:`Access` tab in the :guilabel:`Settings` of
```
[the remark this answers is not in the document]
Comment (comradekingu) [reply docs/admin/access.rst:76]: It could be, but this is the sleeker way (I think) also used elsewhere.
It isn't so much "on the tab" as it is "in the tab", but moreso "beyond the tab on the tab page"?
It was:
"selecting a different access control variation on the Access tab in the Settings of each respective project."

File: docs/admin/access.rst [hunk 18]
```diff
@@ -214,7 +203,7 @@ Managing users and groups
 -------------------------
 
 All users and the various groups they are in can be managed using the
-Django admin interface available, which you can get to by appending
+available Django admin-interface, which you can get to by appending
```
[the remark this answers is not in the document]
Comment (comradekingu) [reply docs/admin/access.rst:206]: It has a deprecated note though. Sounds like it will be removed fully. (Which would remove some complexity.)

File: docs/admin/access.rst (hunk 12 above)
Comment (comradekingu) [reply docs/admin/access.rst:158]: I think that is better, but I don't really know when the check is carried out.

File: docs/admin/access.rst (hunk 11 above)
Comment (comradekingu) [reply docs/admin/access.rst:132]: I think both are valid clauses here.
If I want to remember what an old group did, I can keep the set of privileges it had around.

File: docs/admin/access.rst [hunk 19]
```diff
@@ -79,46 +68,47 @@ Project access control
...
 Custom
-    Django admin manages users instead of Weblate, see :ref:`custom-acl`.
+    The Django admin-interface manages users instead of Weblate, see :ref:`custom-acl`.
 
 .. image:: /images/project-access.png
 
-Grant access to a project by adding the privilege either directly to an user,
-or group of users in the Django admin-interface, or by using user management on
-the project page, as described in :ref:`manage-acl`.
+Grant access to a project by adding the privilege either directly to a user,
```
[the remark this answers is not in the document]
Comment (comradekingu) [reply docs/admin/access.rst:93]: Need a "since version" or tip about that.

File: docs/admin/access.rst [hunk 20]
```diff
@@ -9,62 +9,51 @@ Access control
...
+:ref:`auth-model`.
 
 Site wide permissions
 +++++++++++++++++++++
 
-To manage permissions for a whole instance, just add users to `Users` (this is done
-by default using the :ref:`autogroup`), `Reviewers` and `Managers` groups. Keep
-all projects configured as `Public` (see :ref:`acl`).
+By default, new users get access to comment, provide suggestions, and make translations
+everywhere by being put in the `Users` group automatically by :ref:`autogroup`.
```
[the remark this answers is not in the document]
Comment (comradekingu) [reply docs/admin/access.rst:26]: Will look at this later. Seems there is a dot missing behind "etc."

File: docs/admin/access.rst (hunk 1 above)
Comment (comradekingu) [reply docs/admin/access.rst:39]: It just looked like a general recommendation to me.

File: docs/admin/access.rst [hunk 21]
```diff
@@ -510,3 +499,10 @@ however re-create them if you delete or rename them.
     Never remove the predefined Weblate groups and users, as this can lead to
     unexpected problems. If you have no use for them, you can removing all their
     privileges instead.
+
+Locking down Weblate
+--------------------
+
+If you experience problems with malice on your Weblate installation,
+you can use :setting:`REQUIRE_LOGIN` to require users to sign in and
+:setting:`REGISTRATION_OPEN` to prevent any new registrations.
```
Comment (Fat-Zer) [inline docs/admin/access.rst:508]: I suggest to move this section up again and rename:

```suggestion
Additional access restrictions
--------------------

If you want to use your Weblate installation in a more private manner,
you can configure the Weblate in a such way that only known users will 
have any access to it. In order to do so you should set 
:setting:`REGISTRATION_OPEN` to ``False`` to prevent registrations of any
new users and set :setting:`REQUIRE_LOGIN` to ``/.*`` to require sign in
to access any page.
```

File: docs/admin/access.rst (hunk 15 above)
Comment (Fat-Zer) [reply docs/admin/access.rst:506]: I've looked back into origin of the passage: it was introduced around weblate 1.3; the original intention (as far as I can tell) was to document a way to create a most private weblate installation, see #131.

That's how it ended up in `Common setups`: it's a way to setup the weblate as private as possible, i.e. if you only want to add new users manually and forbid anybody who's unregistered from accessing any page. `Locking down` is kinda misleading here...

See suggestion bellow.

File: docs/admin/access.rst (hunk 18 above)
[the remark this answers is not in the document]
Comment (Fat-Zer) [reply docs/admin/access.rst:206]: It won't be removed soon, I guarantee... Unless some one (namely najel) will reimplement full-featured group/permission management in the weblate UI, but it will likely cause rewrite of most of the access control chapter from scratch yet another time... It's pretty big task which probably won't bring a lot...

File: docs/admin/access.rst (hunk 12 above)
Comment (Fat-Zer) [reply docs/admin/access.rst:158]: ??

the check is performed when a user tries to do a certain action... or what do you mean?

File: docs/admin/access.rst (hunk 11 above)
Comment (Fat-Zer) [reply docs/admin/access.rst:132]: May be a more straightforward variant:
```suggestion
    A role defines a set of permissions, so it can be reused in different groups.
```

File: docs/admin/access.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (Fat-Zer) [reply docs/admin/access.rst:93]: It's from before 3.0 times, there is a `versionchanged` right at the start of the page ;)
Not all the documentation content was changed accordingly since those times...

File: docs/admin/access.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (Fat-Zer) [reply docs/admin/access.rst:85]: Not really =/
I'm a bit confused, why you want to change it at all... It seems perfectly fine to me...

File: docs/admin/access.rst (hunk 20 above)
[the remark this answers is not in the document]
Comment (Fat-Zer) [reply docs/admin/access.rst:26]: yep, I wasn't sure whether `etc` have to end with a dot or not...

File: docs/admin/access.rst (hunk 8 above)
Comment (Fat-Zer) [reply docs/admin/access.rst:102]: For the record, first time I've stumbled upon the term "tallies" in Webblate, I was very confused what is it and felt strong urgent to replace it with `counts`...

I don't suggest to change it... just wanted to share my experience...

File: docs/admin/access.rst (hunk 17 above)
[the remark this answers is not in the document]
Comment (Fat-Zer) [reply docs/admin/access.rst:76]: ok, fine by me...

File: docs/admin/access.rst (hunk 15 above)
[the remark this answers is not in the document]
Comment (comradekingu) [reply docs/admin/access.rst:506]: Well, I really hate it in practice, so I am pretending it is the only reason between the lines.

File: docs/admin/access.rst (hunk 14 above)
Comment (comradekingu) [reply docs/admin/access.rst:28]: @Fat-Zer what I was going for might work better with: "You can put the users you deem fit in groups such as `Users`". Is anything missing then?

File: docs/admin/access.rst (hunk 15 above)
Comment (comradekingu) [reply docs/admin/access.rst:506]: "Ultimate control" is a paradox of insufficiency. It sounds too good.

File: docs/admin/access.rst (hunk 21 above)
Comment (comradekingu) [reply docs/admin/access.rst:508]: I don't think "private" is a good word as it encompasses security. It also isn't the opposite of "public" there.

Comment (Fat-Zer) [reply docs/admin/access.rst:508]: `less public` than? and s/the Weblate/it/ to remove word repetition...

File: docs/admin/access.rst (hunk 15 above)
Comment (Fat-Zer) [reply docs/admin/access.rst:506]: `paradox of insufficiency` ??

**Comment by Fat-Zer** [comment]
@comradekingu, I played a bit more with access control and got some ideas now on how to restructure the whole access control doc so it be relevant to the current state and made some sense in general... I believe I can finish it in a week or so...

I will probably base my version on the current rather than your PR.

Would you mind to put this PR on pause and review/edit the upcoming version rather than this mess of a doc?

**Comment by Fat-Zer** [comment]
@comradekingu, I reworked the whole section almost entirely and restructured it a lot, see #5566.
Be glad if you'll give your review.

**Comment by orangesunny** [comment]
As discussed in the comments of #5566, I am closing this as outdated and going to review and merge #5566. Please read it once it’s live and send a new PR with changes if you feel it’s needed.
